syntax = "proto3";

package ser;

message Vector2 {
    int32 x = 1;
    int32 y = 2;
}

message GameObject {
    repeated Vector2 mesh = 1;
    int32 width = 2;
    int32 height = 3;
    Vector2 position = 4;
    Vector2 velocity = 5;
}

enum PlayerState { 
    IDLE = 0;
    RUN = 1;
    JUMP_UP = 2;
    JUMP_DOWN = 3;
}

message Player{
  GameObject obj  = 1;
  PlayerState state = 2;
  int32 frame = 3;
  bool on_platform = 4;
}

message GameState{
    int32 frame = 1;
    repeated Player players = 2;
    repeated GameObject platforms = 3;
}
